src/app/core/identity-provider/auth0.identity-provider.ts

```diff
@@ -160,6 +108,81 @@ export class Auth0IdentityProvider implements IdentityProvider {
       });
   }
 
+  private normalSignInRegistration(idToken: string) {
+    return this.apiService
+      .post<UserData>('users/processtoken', {
+        id_token: idToken,
+        options: ['CREATE_USER'],
+      })
+      .pipe(
+        tap(() => {
+          this.store.dispatch(loadUserByAPIToken());
+        }),
+        switchMap((userData: UserData) =>
+          combineLatest([this.store.pipe(select(getLoggedInCustomer)), this.store.pipe(select(getUserLoading))]).pipe(
+            filter(([, loading]) => !loading),
+            first(),
+            switchMap(([customer]) =>
+              !customer
+                ? this.router.navigate(['/register', 'sso'], {
+                    queryParams: {
+                      userid: userData.businessPartnerNo,
+                      firstName: userData.firstName,
+                      lastName: userData.lastName,
+                    },
+                  })
+                : of(false)
+            ),
+            switchMap((navigated: boolean) =>
+              navigated || navigated === null
+                ? race(
+                    this.store.pipe(
+                      select(getSsoRegistrationRegistered),
+                      whenTruthy(),
+                      tap(() => {
+                        this.store.dispatch(loadUserByAPIToken());
+                      })
+                    ),
+                    this.store.pipe(
+                      select(getSsoRegistrationCancelled),
+                      whenTruthy(),
+                      mapTo(false),
+                      tap(() => this.router.navigateByUrl('/logout'))
+                    )
+                  )
+                : of(navigated)
+            )
+          )
+        ),
+        switchMapTo(this.store.pipe(select(getUserAuthorized), whenTruthy(), first())),
+        catchError((error: HttpError) => {
+          this.apiTokenService.removeApiToken();
+          this.triggerLogout();
+          return of(error);
+        })
+      );
+  }
+
+  private inviteRegistration(idToken: string, userId: string, hash: string) {
+    window.sessionStorage.removeItem('invite-userid');
+    window.sessionStorage.removeItem('hash');
+    return this.apiService
+      .post<UserData>('users/processtoken', {
+        id_token: idToken,
+        secure_user_ref: {
+          user_id: userId,
+          secure_code: hash,
+        },
+        options: ['UPDATE'],
+      })
+      .pipe(
+        tap(() => {
+          this.store.dispatch(loadUserByAPIToken());
+        }),
+        switchMapTo(this.store.pipe(select(getUserAuthorized), whenTruthy(), first()))
```

Comment:
I cannot see any kind of error handling. What happens if above API call is returning with something other than 2xx?

Reply:
you're right, I added some error handling like in the normal sign in case.